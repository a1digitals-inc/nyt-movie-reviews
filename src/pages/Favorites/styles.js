import styled from 'styled-components';

export const Container = styled.div`
  max-width: 1020px;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  padding: 0 20px 50px;
`;
