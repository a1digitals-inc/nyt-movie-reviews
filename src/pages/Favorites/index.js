import React from 'react';

import Review from '../../components/Review';

import { Container } from './styles';

export default function Reviews() {
  return (
    <Container>
      <Review
        ReviewTitle="The Irishman"
        ReviewDescription="Robert De Niro, Al Pacino and Joe Pesci star in Martin Scorsese’s monumental, elegiac tale of violence, betrayal, memory and loss. It’s the opening-night movie at the New York Film Festival."
        ReviewDate={new Date()}
        ReviewImage="https://static01.nyt.com/images/2019/09/27/arts/27irishman1/27irishman1-mediumThreeByTwo210.jpg"
        ReviewUrl="#url"
        CriticName="Critic name"
        CriticsPick
      />

      <Review
        ReviewTitle="The Irishman"
        ReviewDescription="Robert De Niro, Al Pacino and Joe Pesci star in Martin Scorsese’s monumental, elegiac tale of violence, betrayal, memory and loss. It’s the opening-night movie at the New York Film Festival."
        ReviewDate={new Date()}
        ReviewImage="https://static01.nyt.com/images/2019/09/27/arts/27irishman1/27irishman1-mediumThreeByTwo210.jpg"
        ReviewUrl="#url"
        CriticName="Critic name"
      />
    </Container>
  );
}
