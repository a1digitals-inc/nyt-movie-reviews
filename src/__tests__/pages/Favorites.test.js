import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Favorites from '../../pages/Favorites';

describe('Favorites page', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    favorites: [],
  };
  const store = mockStore(initialState);

  it('should render without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Favorites />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
