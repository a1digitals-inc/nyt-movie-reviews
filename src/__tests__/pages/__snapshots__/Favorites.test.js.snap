// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites page should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Reviews>
    <NotFound
      Code={null}
      Message="No records found"
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "cOsgYx",
                "rules": Array [
                  "
  display: flex;
  justify-content: center;
  align-items: center;
  font-style: oblique;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="sc-bdVaJa cOsgYx"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "fMMdfB",
                      "rules": Array [
                        "
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-top: 90px;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="sc-bwzfXH fMMdfB"
                >
                  <strong />
                  <strong>
                    No records found
                  </strong>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </NotFound>
  </Reviews>
</Provider>
`;
